### DISCUSSION TITLE
Search OR operator, search by filesystem path?

### DISCUSSION THREAD

**Discussion by hjri** [jcorporation/myMPD#689] [Q&A]
I want to filter my library to show say all soundtrack albums, however not every album has "Soundtrack" in its name, some use OST/O.S.T. and some just use name of the game/series, "Soundtrack" is also not really a genre of music. It doesn't seem to be possible to use logical OR operator when filtering, replacing AND with OR makes a weird unusable query.

Luckily i store my soundtracks in a separate folder but it doesn't seem to be able to filter by path either... am i missing something?

**Comment by jcorporation** [comment]
Right, there is no or operator or search by filename/filepath for albums. OR operator is not supported by MPD and the album search is only based on tags.

There are 2 possibilities to achieve your requirement:
1. add multiple genre tags to your music files
2. use a regex (=~) like `(ost|o\.s\.t\.|soundtrack)`

**Reply by hjri** [reply]
would it be possible to add option to search by file path as well?

regex could work (although i didn't know it's using that) but how would i add search for say Album =~ (ost|sound ?track) OR Genre = Soundtrack

It would help tremendously to have arbitrary tag system in or on top of MPD, i.e. being able to tag songs as `soundtrack` `videogame` `synthwave` etc without trying it down to genre field

**Comment by Emilot** [comment]
The integration of Musicbrain will help these issues JC, right?

i hope... :)

**Comment by jcorporation** [comment]
Filepath filtering should be possible.

@Emilot: no MusicBrainz integration does not solve this issue. 

I tag my files with MusicBrainz Picard. MBIDs will be used for MusicBrainz integration.

**Comment by Emilot** [comment]
Thanx JC
